core-web/libs/edit-content/src/lib/fields/dot-edit-content-category-field/components/dot-category-field-category-list/dot-category-field-category-list.component.ts

```diff
@@ -0,0 +1,127 @@
+import { CommonModule } from '@angular/common';
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    Component,
+    computed,
+    DestroyRef,
+    effect,
+    ElementRef,
+    EventEmitter,
+    inject,
+    input,
+    Output,
+    QueryList,
+    ViewChildren
+} from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { FormsModule } from '@angular/forms';
+
+import { ButtonModule } from 'primeng/button';
+import { CheckboxModule } from 'primeng/checkbox';
+import { TreeModule } from 'primeng/tree';
+
+import { DotCategory } from '@dotcms/dotcms-models';
+
+import { DotCategoryFieldCategory } from '../../models/dot-category-field.models';
+
+export const MINIMUM_CATEGORY_COLUMNS = 4;
+
+const MINIMUM_CATEGORY_WITHOUT_SCROLLING = 3;
+
+/**
+ * Represents the Dot Category Field Category List component.
+ * @class
+ * @implements {AfterViewInit}
+ */
+@Component({
+    selector: 'dot-category-field-category-list',
+    standalone: true,
+    imports: [CommonModule, TreeModule, CheckboxModule, FormsModule, ButtonModule],
+    templateUrl: './dot-category-field-category-list.component.html',
+    styleUrl: './dot-category-field-category-list.component.scss',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    // eslint-disable-next-line @angular-eslint/no-host-metadata-property
+    host: {
+        class: 'category-list__wrapper'
+    }
+})
+export class DotCategoryFieldCategoryListComponent implements AfterViewInit {
+    @ViewChildren('categoryColumn') categoryColumns: QueryList<ElementRef>;
+
+    /**
+     * Represents the variable 'categories' which is of type 'DotCategoryFieldCategory[][]'.
+     */
+    categories = input.required<DotCategoryFieldCategory[][]>();
+    /**
+     * Represent the selected item saved in the contentlet
+     */
+    selected = input.required<string[]>();
+
+    /**
+     * Generate the empty columns
+     */
+    emptyColumns = computed(() => {
+        const numberOfEmptyColumnsNeeded = Math.max(
+            MINIMUM_CATEGORY_COLUMNS - this.categories().length,
+            0
+        );
+
+        return Array(numberOfEmptyColumnsNeeded).fill(null);
+    });
+
+    /**
+     * Emit the item clicked to the parent component
+     */
+    @Output() itemClicked = new EventEmitter<{ index: number; item: DotCategory }>();
```

Comment:
@oidacra If we already use signal inputs, why not use signal outputs?

Reply:
Because, we use Angular 17.1, the output comes in 17.3 :(